packages/header/src/Header.tsx

```diff
@@ -0,0 +1,225 @@
+
+const Icons = styled.div`
+  display: flex;
+  align-items: center;
+  flex-direction: row-reverse;
+
+  > * {
+    margin-left: 40px;
+  }
+`
+
+const AppSelectorWrapper = styled.div`
+  position: absolute;
+  top: 60px;
+  left: 0;
+  min-width: 300px;
+  max-width: 300px;
+  background: #ffffff;
+  border: 1px solid gray;
+  display: flex;
+  flex-wrap: wrap;
+  flex-direction: row;
+  padding: 10px;
+  width: min-content;
+`
+
+const AppBox = styled.div`
+  border: 3px solid grey;
+  padding: 8px;
+  margin: 5px;
+  height: 80px;
+  width: 80px;
+  background: #b3dae0;
+  color: black;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  cursor: pointer;
+
+  &:hover {
+    background: #4f878d;
+  }
+`
+const ClickableIcon = styled.div`
+  &:hover {
+    color: gray;
+    cursor: pointer;
+  }
+`
+
+const UserInfoLabel = styled.b`
+  margin-left: 5px;
+`
+
+const Row = styled.div`
+  display: flex;
+  flex-direction: row;
+`
+
+const UnstyledList = styled.ul`
+  margin: 0;
+  padding: 0;
+  list-style-type: none;
+  display: inline-flex;
+`
+
+type THeaderPluginConfig = {
+  userInfo: boolean
+  about: boolean
+}
+
+type TApplication = {
+  name: string
+  dataSources: string[]
+  label: string
+  description: string
+  roles: string[]
+  adminRole: string
+}
+
+export default (props: IUIPlugin): JSX.Element => {
+  const { idReference, config: passedConfig } = props
+  const config: THeaderPluginConfig = passedConfig
+  const [entity, isLoading] = useDocument<TApplication>(idReference)
+  const { tokenData, token, logOut } = useContext(AuthContext)
+  const [aboutOpen, setAboutOpen] = useState(false)
+  const [visibleUserInfo, setVisibleUserInfo] = useState<boolean>(false)
+  const [appSelectorOpen, setAppSelectorOpen] = useState<boolean>(false)
+  const [apiKey, setAPIKey] = useState<string | null>(null)
+  const dmssApi = new DmssAPI(token)
+  const [checked, updateChecked] = useLocalStorage<string | null>(
+    'impersonateRoles',
+    null
+  )
+
+  if (isLoading || !entity) {
+    return <Loading />
+  }
+
+  return (
+    <TopBar>
+      <TopBar.Header>
+        {/*<ClickableIcon onClick={() => setAppSelectorOpen(!appSelectorOpen)}>*/}
+        <Icon name="grid_on" size={32} />
+        {/*</ClickableIcon>*/}
+        <h4 style={{ paddingTop: 9, paddingLeft: 10 }}>{entity.label}</h4>
+        {/*{appSelectorOpen && (*/}
+        {/*  <AppSelectorWrapper>*/}
+        {/*    {entity.dataSources.map((app) => (*/}
+        {/*      <Link to={`/${app.urlPath}`} key={app.name}>*/}
+        {/*        <AppBox>{app?.label ? app.label : app.name}</AppBox>*/}
+        {/*      </Link>*/}
+        {/*    ))}*/}
+        {/*    <Link to={'/DMT/search'}>*/}
+        {/*      <AppBox>Search</AppBox>*/}
+        {/*    </Link>*/}
+        {/*  </AppSelectorWrapper>*/}
+        {/*)}*/}
+      </TopBar.Header>
+      <TopBar.Actions>
+        <Icons>
+          <ClickableIcon onClick={() => setAboutOpen(true)}>
```

Comment:
I miss logic solving the issue of both dialogs being open at once.

Reply:
Clicking outside the Dialog closes the first one automatically